libnavigation-base/src/test/java/com/mapbox/navigation/base/route/NavigationRouteWaypointsTest.kt

```diff
@@ -0,0 +1,157 @@
+                    emptyList<DirectionsWaypoint>(),
+                    emptyList<DirectionsWaypoint>(),
+                    emptyList<DirectionsWaypoint>()
+                ),
+                arrayOf(
+                    true,
+                    emptyList<DirectionsWaypoint>(),
+                    emptyList<DirectionsWaypoint>(),
+                    emptyList<DirectionsWaypoint>()
+                ),
+                // #15
+                arrayOf(
+                    null,
+                    emptyList<DirectionsWaypoint>(),
+                    filledWaypoints1,
+                    emptyList<DirectionsWaypoint>()
+                ),
+                arrayOf(
+                    false,
+                    emptyList<DirectionsWaypoint>(),
+                    filledWaypoints1,
+                    emptyList<DirectionsWaypoint>()
+                ),
+                arrayOf(true, emptyList<DirectionsWaypoint>(), filledWaypoints1, filledWaypoints1),
+                // #18
+                arrayOf(null, filledWaypoints2, null, filledWaypoints2),
+                arrayOf(false, filledWaypoints2, null, filledWaypoints2),
+                arrayOf(true, filledWaypoints2, null, null),
+                // #21
+                arrayOf(null, filledWaypoints2, emptyList<DirectionsWaypoint>(), filledWaypoints2),
+                arrayOf(false, filledWaypoints2, emptyList<DirectionsWaypoint>(), filledWaypoints2),
+                arrayOf(
+                    true,
+                    filledWaypoints2,
+                    emptyList<DirectionsWaypoint>(),
+                    emptyList<DirectionsWaypoint>()
+                ),
+                // #24
+                arrayOf(null, filledWaypoints2, filledWaypoints1, filledWaypoints2),
+                arrayOf(false, filledWaypoints2, filledWaypoints1, filledWaypoints2),
+                arrayOf(true, filledWaypoints2, filledWaypoints1, filledWaypoints1),
+            )
+        }
+    }
+
+    @Test
+    fun waypoints() {
+        val route = NavigationRoute(
+            DirectionsResponse.builder()
+                .routes(
+                    listOf(
+                        DirectionsRoute.builder()
+                            .distance(1.0)
+                            .duration(2.9)
+                            .waypoints(routeWaypoints)
+                            .build()
+                    )
+                )
+                .waypoints(responseWaypoints)
+                .code("Ok")
+                .build(),
+            0,
+            RouteOptions.builder()
+                .profile(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC)
+                .coordinatesList(
+                    listOf(
+                        Point.fromLngLat(1.1, 2.2),
+                        Point.fromLngLat(3.3, 4.4)
+                    )
+                )
+                .waypointsPerRoute(waypointsPerRoute)
+                .build(),
+            mockk(relaxed = true)
+        )
```

Comment:
would `com.mapbox.navigation.testing.factories.NavigationRouteFactoryKt#createNavigationRoutes` simplify this setup?